packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_loan_create_vault.test.ts

```diff
@@ -0,0 +1,293 @@
+      isUnderLiquidation: false,
+      collateralAmounts: [],
+      loanAmount: [],
+      collateralValue: expect.any(BigNumber),
+      loanValue: expect.any(BigNumber),
+      currentRatio: expect.any(BigNumber)
+    })
+  })
+
+  it('should createVault with the default scheme if the given schemeId is empty', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.createVault({
+      ownerAddress: script,
+      schemeId: ''
+    }, script)
+
+    // Ensure the created txn is correct
+    const outs = await sendTransaction(testing.container, txn)
+    expect(outs[0].value).toStrictEqual(0)
+    expect(outs[1].value).toBeLessThan(10)
+    expect(outs[1].value).toBeGreaterThan(9.999)
+    expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+    // Ensure you don't send all your balance away
+    const prevouts = await providers.prevout.all()
+    expect(prevouts.length).toStrictEqual(1)
+    expect(prevouts[0].value.toNumber()).toBeLessThan(10)
+    expect(prevouts[0].value.toNumber()).toBeGreaterThan(9.999)
+
+    const txid = calculateTxid(txn)
+
+    await testing.generate(1)
+    const data = await testing.rpc.call('getvault', [txid], 'bignumber')
+    expect(data).toStrictEqual({
+      loanSchemeId: 'default',
+      ownerAddress: await providers.getAddress(),
+      isUnderLiquidation: false,
+      collateralAmounts: [],
+      loanAmount: [],
+      collateralValue: expect.any(BigNumber),
+      loanValue: expect.any(BigNumber),
+      currentRatio: expect.any(BigNumber)
+    })
+  })
+
+  it('should createVault and then again createVault with the same parameters', async () => {
+    const script = await providers.elliptic.script()
+    const txn = await builder.loans.createVault({
+      ownerAddress: script,
+      schemeId: 'scheme2'
+    }, script)
+
+    // Ensure the created txn is correct
+    const outs = await sendTransaction(testing.container, txn)
+    expect(outs[0].value).toStrictEqual(0)
+    expect(outs[1].value).toBeLessThan(10)
+    expect(outs[1].value).toBeGreaterThan(9.999)
+    expect(outs[1].scriptPubKey.addresses[0]).toStrictEqual(await providers.getAddress())
+
+    // Ensure you don't send all your balance away
+    const prevouts = await providers.prevout.all()
+    expect(prevouts.length).toStrictEqual(1)
+    expect(prevouts[0].value.toNumber()).toBeLessThan(10)
+    expect(prevouts[0].value.toNumber()).toBeGreaterThan(9.999)
+
+    const txid = calculateTxid(txn)
+
+    await testing.generate(1)
+    const data = await testing.rpc.call('getvault', [txid], 'bignumber')
+    expect(data).toStrictEqual({
+      loanSchemeId: 'scheme2',
+      ownerAddress: await providers.getAddress(),
+      isUnderLiquidation: false,
+      collateralAmounts: [],
+      loanAmount: [],
+      collateralValue: expect.any(BigNumber),
+      loanValue: expect.any(BigNumber),
+      currentRatio: expect.any(BigNumber)
+    })
```

Comment:
```suggestion
```

Maybe redundant as you test it in the first test and here we just want to check the second vault creation ?

Reply:
The test demonstrates that even though you use the same parameters, calling a second time would create a different vault with the same properties.